skyportal/handlers/api/annotation_services.py

```diff
@@ -146,3 +147,133 @@ def post(self, obj_id):
+                      Default is VII/290.
+                  crossmatchRadius:
+                    required: false
+                    type: number
+                    description: |
+                      Crossmatch radius (in arcseconds) to retrieve photoz's
+                      Default is 2.
+                  group_ids:
+                    required: false
+                    type: array
+                    items:
+                      type: integer
+                    description: |
+                      List of group IDs corresponding to which groups
+                      should be able to view annotation.
+                      Defaults to all of requesting user's groups.
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        obj = Obj.query.get(obj_id)
+        if obj is None:
+            return self.error('Invalid object id.')
+
+        data = self.get_json()
+
+        group_ids = data.pop('group_ids', None)
+        if not group_ids:
+            groups = self.current_user.accessible_groups
+        else:
+            try:
+                groups = Group.get_if_accessible_by(
+                    group_ids, self.current_user, raise_if_none=True
+                )
+            except AccessError:
+                return self.error(
+                    'At least some of the groups are not accessible.', status=403
+                )
+
+        author = self.associated_user_object
+
+        catalog = data.pop('catalog', "VII/290")
+        radius_arcsec = data.pop('crossmatchRadius', 2.0)
+        candidate_coord = SkyCoord(ra=obj.ra * u.deg, dec=obj.dec * u.deg)
+
+        tl = Vizier.query_region(
+            coordinates=candidate_coord,
+            catalog=catalog,
+            radius=radius_arcsec * u.arcsec,
+        )
+
+        if len(tl) == 0:
+            return self.error("No successful cross-match available.")
+
+        if len(tl) > 1:
+            return self.error("Should only have one table from that query.")
+        df = tl[0].filled(fill_value=-99).to_pandas()
+        keys = [
+            'Qpct',
+            'z',
+        ]
+        annotations = []
+        for index, row in df.iterrows():
+            annotation_data = {
+                k: row.to_dict().get(k, None)
+                for k in keys
+                if not row.to_dict().get(k, None) == -99
+            }
+
+            origin = f"{catalog}-{row['Name']}"
+            annotation = Annotation(
+                data=annotation_data,
+                obj_id=obj_id,
+                origin=origin,
+                author=author,
+                groups=groups,
+            )
+            annotations.append(annotation)
+
+        if len(annotations) == 0:
+            return self.error("No QSO flag available.")
```

Comment:
I'm not sure how QSO flag is related to this.

Reply:
me neither.